### ISSUE TITLE
[Bug]: player.seeEnemy() cannot see enemies hidden by attack components

### ISSUE THREAD

**Issue by CubeDr** [RafaelBarbosatec/bonfire#455]
### What happened?

I'm auto-targeting the enemies and firing range attacks.

Since attacks are `FlyingAttackGameObject`, they hide the enemies and the player cannot see the enemy when the attack is flying to the enemy.

### Steps to reproduce?

1. Find the enemy with `seeEnemy()`
2. Get the angle with `angleTo()`
3. fire range attack with `simpleAttackRangeByAngle()`

```Markdown
@override
void fire(Player player, Weapon weapon) async {
  if (player.isDead) return;

  player.simpleAttackRangeByAngle(
    speed: 200,
    angle: await _getFireAngle(player) - pi / 2,
    damage: _damage,
    animation: AnimationConfigs.shurikenMagicAnimation(),
    size: _size,
    attackFrom: AttackFromEnum.PLAYER_OR_ALLY,
    collision: CircleHitbox(
      radius: _radius,
      position: Vector2.zero(),
      collisionType: CollisionType.passive,
    ),
  );
}
Future<double> _getFireAngle(Player player) {
  final completer = new Completer<double>();
  player.seeEnemy(
    radiusVision: 2000,
    observed: (List<Enemy> enemies) {
      double distance = double.infinity;
      Enemy closetEnemy = enemies[0];
      for (final enemy in enemies) {
        double d = player.distance(enemy);
        if (d < distance) {
          distance = d;
          closetEnemy = enemy;
        }
      }
      completer.complete(player.angleTo(closetEnemy.position));
    },
  );
  return completer.future;
}
```

### What did you expect to happen?

Only fire range attacks to the closest enemy.

### Bonfire version

Release 3.0.0

**Comment by CubeDr** [comment]
I was able to bypass the issue by disabling the raycast check with `setupVision(checkWithRaycast: false);`.

However, I think having an option to prevent projectiles from blocking the sight will greatly improve the framework.

Thanks, bonfire team!

**Comment by RafaelBarbosatec** [comment]
Fixed in version `3.0.10`
